### PR TITLE
Add parameter for /v1/ip_info to include company ID providers

### PR DESCRIPTION
**Pull request by limemloh**
## Purpose

Optionally provide the company ID providers, in a backwards compatible way, allowing wallets to only include them when Company IDs are fully supported.

## Changes

- Introduce `/v2/ip_info` endpoint which include Company ID providers.
- The service takes an additional file, which is the file to serve when Company ID providers are to be included.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -69,7 +70,8 @@ parser =
...
             <*> optional (option auto (long "health-tolerance" <> metavar "SECONDS" <> help "the maximum tolerated age of the last final block in seconds before the health query returns false."))
             <*> strOption (long "ip-data" <> metavar "FILE" <> help "File with public and private information on the identity providers, together with metadata.")
-            <*> strOption (long "ip-data-v1" <> metavar "FILE" <> help "File with public and private information on the identity providers for the flow without initial accounts, together with metadata.")
+            <*> strOption (long "ip-data-v1" <> metavar "FILE" <> help "File with public and private information on the identity providers (excluding Company ID providers) for the flow without initial accounts, together with metadata.")
+            <*> strOption (long "ip-data-company-v1" <> metavar "FILE" <> help "File with public and private information on the identity providers (including Company ID providers) for the flow without initial accounts, together with metadata.")
```
Comment (td202) [inline app/Main.hs:74]: Also update the README.md with this option.

Comment (limemloh) [reply app/Main.hs]: Done

File: ChangeLog.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased changes
 
+## 0.31.0
+
+- Introduce `/v2/ip_info` endpoint which include Company ID providers, the information is provided to the service using `--ip-data-v2 <FILE>`.
```
Comment (td202) [inline ChangeLog.md:7]:
```suggestion
- Introduce `/v2/ip_info` endpoint which includes Company ID providers. The information is provided to the service using `--ip-data-v2 <FILE>`.
```

File: README.md [hunk 3]
```diff
@@ -20,8 +20,9 @@ The wallet proxy provides the following endpoints:
 * `PUT /v0/testnetGTUDrop/{account address}`: request a CCD drop to the specified account
 * `GET /v0/health`: get a response specifying if the wallet proxy is up to date
 * `GET /v0/global`: get the cryptographic parameters obtained from the node it is connected to
-* `GET /v0/ip_info`: get the identity providers information, including links for
-  submitting initial identity issuance requests.
+* `GET /v0/ip_info`: get the identity providers information with links for submitting initial identity issuance requests.
+* `GET /v1/ip_info`: get the identity providers information with links for submitting identity issuance and recovery requests.
+* `GET /v2/ip_info`: get the identity providers information (including company ID providers) with for submitting identity issuance and recovery requests.
```
Comment (td202) [inline README.md:25]: Please also update the Deployment section.

**Review by td202 — APPROVED** [review]
Looks good.

**Pull request merged**